proto/iotics/api/meta.proto

```diff
@@ -0,0 +1,125 @@
+
+
+// MetaAPI enables querying of metadata associated with Twins and Feeds.
+service MetaAPI {
+
+  // SparqlQuery performs a SPARQL 1.1 query and returns one or more results, each as a sequence of chunks. Note that:
+  // - Chunks for a particular result will arrive in-order though they might be interleaved with chunks from other
+  //   results (when performing a non-local query). See scope parameter in SparqlQueryRequest.
+  // - The call will only complete once the (specified or host default) request timeout has been reached. The client can
+  //   choose to end the stream early once they have received enough results. (E.g. in the case of Scope.LOCAL this
+  //   would be after the one and only sequence of chunks has been received.)
+  rpc SparqlQuery(SparqlQueryRequest) returns (stream SparqlQueryResultChunk) {}
+}
+
+// Available result content types for SPARQL requests. Note that applicable content types depend on the type of query.
+// Specifying an incompatible result type can lead to the result being returned with a different content type.
+enum SparqlResultType {
+
+  // Applicable to SELECT/ASK (SPARQL 1.1 Query Results JSON Format)
+  SPARQL_JSON = 0;
+  // Applicable to SELECT/ASK (SPARQL 1.1 Query Results XML Format)
+  SPARQL_XML = 1;
+  // Applicable to SELECT/ASK (SPARQL 1.1. Query Results CSV Format)
+  SPARQL_CSV = 2;
+  // Applicable to CONSTRUCT/DESCRIBE (Terse RDF Triple Language)
+  RDF_TURTLE = 3;
+  // Applicable to CONSTRUCT/DESCRIBE (RDF 1.1 XML)
+  RDF_XML = 4;
+  // Applicable to CONSTRUCT/DESCRIBE (RDF 1.1 N-Triples)
+  RDF_NTRIPLES = 5;
+  // Applicable to CONSTRUCT/DESCRIBE (RDF 1.1 N-Quads)
+  RDF_NQUADS = 6;
+}
+
+// SparqlQueryRequest describes a SPARQL query.
+message SparqlQueryRequest {
+
+  // SPARQL request payload.
+  message Payload {
+
+    // The desired result content type. Note that the actual content (mime) type could be different and will be returned
+    // in result chunks.
+    SparqlResultType resultTypeHint = 1;
+    // A UTF8-encoded SPARQL 1.1 query
+    bytes query = 2;
+  }
+
+  // SPARQL request headers
+  Headers headers = 1;
+
+  // SPARQL request scope
+  Scope scope = 2;
+
+  // SPARQL request payload
+  Payload payload = 3;
+}
+
+// Part of a result for a SPARQL query request. Multiple chunks form a complete result. Related chunks can be identified
+// by a combination of:
+// - The host ID (unset for local results)
+// - Client reference (in headers, set by caller)
+// - Chunk sequence number
+// Note that chunks are guaranteed to arrive in-order (for a specific result from a host) but could be duplicated. It is
+// up to the client to ignore duplicates (based on sequence number).
+message SparqlQueryResultChunk {
+
+  // Payload of the query result chunk
+  message Payload {
+
+    // Result host identifier. Indicates from which host this result chunk came from. For a local result, this field
+    // will be unset.
+    HostID remoteHostId = 1;
```

Comment:
Not sure I like the reliance on absence of value to indicate a local host value.

Seems like this might simplify code in multiple places to just always set the value and construct responses in a similar fashion for local/remote.

Reply:
> to just always set the value

Can you explain what you mean? Do you mean set it to the local host ID?

The suggested behaviour is the same as for search currently (and also discover/follow - where absence/nil value means it's a local twin/feed).

The client is not aware of the local host ID (without performing additional lookups) - so if it were set - they'd have to

Reply:
That's something that has been raised several times about the search. It will be easier to make something consistent when we will replace the `remoteHostId`  (peerID) with the host twin DID. We don't necessarily need to address that right now since the work to return the hosttwinDID is not trivial but it is great to keep that in mind